File: module-persistence/src/main/java/com/hanghae/module/persistence/entity/TheaterEntity.java
package com.hanghae.module.persistence.entity;

import com.hanghae.module.common.audit.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "theater")
@Builder
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class TheaterEntity extends BaseEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String name;
}

File: domain/src/main/kotlin/com/hanghe/redis/theater/TheaterEntity.kt
package com.hanghe.redis.theater

import com.hanghe.redis.BaseEntity
import com.hanghe.redis.movie.seat.SeatEntity
import com.hanghe.redis.screening.ScreeningEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany

@Entity(name = "theaters")
class TheaterEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    val name: String,

    @OneToMany(mappedBy = "theater")
    val screenings: Set<ScreeningEntity> = emptySet(),

    @OneToMany(mappedBy = "theater")
    val seats: Set<SeatEntity> = emptySet()
) : BaseEntity()
